src/types.md

```diff
@@ -0,0 +1,183 @@
+# Questions about your types
+
+## My 'class' needs references to lots of other things in order to do its job, and that conflicts with other things which also hold similar references. What do I do?
+
+It's common in C++ to have objects which need to mutate other objects to get
+their work done. Those other objects may be accessed by several different
+objects.
+
+```mermaid
+flowchart LR
+    subgraph Shared functionality
+    important[Important Shared Object]
+    end
+    subgraph ObjectA
+    methodA[Method]
+    refa[Mutable Reference]-->important
+    methodA-. Acts on shared object.->important
+    end
+    subgraph ObjectB
+    refb[Mutable Reference]-->important
+    methodB[Method]
+    methodB-. Acts on shared object.->important
+    end
+    main --> ObjectA
+    main --> ObjectB
+    main-. Calls .-> methodA
+    main-. Calls .-> methodB
+```
+
+You can't have multiple mutable references to a shared object, so what do you do?
+
+First of all, consider moving behavior out of your types. (See
+[the answer about the observer pattern](./codebase.md#the-c-observer-pattern-is-hard-in-rust-what-to-do) and especially
+[the second option described there](./codebase.md#option-2-drive-the-objects-from-the-code-not-the-other-way-round).)
+
+Even in Rust, though, it's still often the best choice to have complex
+behavior as part of types within `impl` blocks. You can still do that -
+but don't _store_ references. Instead, pass them into each function call.
+
+```mermaid
+flowchart LR
+    subgraph Shared functionality
+    important[Important Shared Object]
+    end
+    subgraph ObjectA
+    methodA[Method]
+    methodA-. Acts on shared object.->important
+    end
+    subgraph ObjectB
+    methodB[Method]
+    methodB-. Acts on shared object.->important
+    end
+    main --> ObjectA
+    main --> ObjectB
+    main --> important
+    main-. Passes reference to shared object.-> methodA
+    main-. Passes reference to shared object.-> methodB
+```
+
+Instead of:
```

Comment:
Suggest adding "this". That gives you more consistency between "Instead of this:" and "Do this:"